env0/configuration_template.go

```diff
@@ -59,10 +59,10 @@ func getConfigurationTemplateSchema(templateType TemplateType) map[string]*schem
 			},
 		},
 		"gitlab_project_id": {
-			Type:         schema.TypeInt,
-			Description:  "the project id of the relevant repository",
-			Optional:     true,
-			RequiredWith: []string{"token_id"},
+			Type:        schema.TypeInt,
+			Description: "the project id of the relevant repository (deprecated)",
+			Deprecated:  "'repository' is used instead",
```

Comment:
Text should be changed
We should use "project id is now auto-fetched from the repository URL"